src/ompl/base/src/ProblemDefinition.cpp

```diff
@@ -425,8 +425,10 @@ void ompl::base::ProblemDefinition::addSolutionPath(const PathPtr &path, bool ap
                                                     const std::string &plannerName) const
 {
     PlannerSolution sol(path);
-    Cost sol_cost {path->cost(optimizationObjective_)};
-    sol.setOptimized(optimizationObjective_, sol_cost, false);
+    if (!computeSolutionCost_) {
```

Comment:
logging useful here to know if this code path has/hasn't been executed

Reply:
addressed